Create configure_syslog_ssl (#296)
IBM/universal-connectors: docs/configure_syslog_ssl_failover.md

# Configuring SSL with syslog

> Syslog for universal connector is based on Logstash's tcp input plug-in. In order to enable TCP with SSL, first use the API to generate an SSL certificate. Then, modify the rsyslog configuration file on the database server (MySQL or MongoDB) with the certificate details and update the connector configuration on the collector.

Reviewer: ```suggestion
Syslog for universal connector is based on Logstash's TCP input plug-in. In order to enable TCP with SSL, first use the API to generate an SSL certificate. Then, modify the rsyslog configuration file on the database server (MySQL or MongoDB) with the certificate details and update the connector configuration on the collector.
```

            global(DefaultNetstreamDriverCAFile="/path/to/logstash.crt")
            
            # set up the action for all messages
            action(type="omfwd" protocol="tcp" port="6514"
                   StreamDriver="gtls" StreamDriverMode="1" StreamDriverAuthMode="anon")
            
            # In this example, transmitting the data is done via the imfile module with mongod
            $FileCreateMode 0640
            
            
            module(load="imfile")
            
            input(type="imfile" Tag="mongod: " File="/path/to/mongo_audit_rsyslog_buffer_file" ruleset="pRuleset")
            
            
            ruleset(name="pRuleset") {
            action(type="omfwd"
                keepalive="on"
                    protocol="tcp"
                    target="<COLLECTOR_IP>"
                    port="<COL_PORT>"
               StreamDriver="gtls" StreamDriverMode="1" StreamDriverAuthMode="anon")
            }
      ```
      
Where `/path/to` is the path to the certificate file that was copied to the database server in step 2.

**Note:** `/path/to/mongo_audit_rsyslog_buffer_file` is the path defined in mongod.conf for routing audit logs

> **Note:** more `ruleset`s can be added specifying more collectors if failover is required. Otherwise, the there will be event duplication.

Reviewer: ```suggestion
**Note:** more `ruleset`s can be added specifying more collectors if failover is required. Otherwise, there will be event duplication.
```

### Example with failover enabled:

```
ruleset(name="pRuleset") {
 action(type="omfwd"
                 keepalive="on"
                 protocol="tcp"
                 target="primary collector IP"
                 port="5000")
 action(type="omfwd"
         keepalive="on"
         protocol="tcp"
         target="Secondary collector IP"
         port="5000"
         action.execOnlyWhenPreviousIsSuspended= "on")
 action(type="omfile"
 file="Path_TO_Local_BUFFER_FILE/mongo_audit_rsyslog_buffer" ## -->LOCAL machine
 action.execOnlyWhenPreviousIsSuspended= "on")
}
if($programname contains "mongod") then {
    call pRuleset
}
```

> 4. Enable UC and add a connector with MongoDB (MySQL) with Syslog template from dropdown menu

Reviewer: ```suggestion
4. Enable the universal connector and add a connector with MongoDB (MySQL) with Syslog template from the drop-down menu.
```
